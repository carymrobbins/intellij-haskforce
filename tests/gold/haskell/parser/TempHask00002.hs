{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE QuasiQuotes       #-}
{-# LANGUAGE TemplateHaskell   #-}
module TempHask00002 where

[undefined|
something here "hello world"
|]

[| 7 |]

[d| 7 |]

[t| Int |]

[p| 7 |]

$(deriveStuff 'f)
m = $x

''T

f n = \ [haskell|y|] -> y+n
