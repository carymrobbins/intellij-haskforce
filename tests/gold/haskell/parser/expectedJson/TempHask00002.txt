[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":24,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":24,"srcSpanEndLine":24,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":24,"srcSpanEndLine":24,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":8},"srcInfoPoints":[]},"TempHask00002"]},null,null]},[{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":14},"srcInfoPoints":[]},"FlexibleInstances"]}]]},{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":14},"srcInfoPoints":[]},"QuasiQuotes"]}]]},{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":14},"srcInfoPoints":[]},"TemplateHaskell"]}]]}],[],[{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":6,"srcSpanEndLine":8,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"QuasiQuote":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":6,"srcSpanEndLine":8,"srcSpanStartColumn":1},"srcInfoPoints":[]},"undefined","\nsomething here \"hello world\"\n"]}]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":6}]},{"BracketExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":6}]},{"ExpBracket":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":4},"srcInfoPoints":[]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":4},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":4},"srcInfoPoints":[]},7,"7"]}]}]}]}]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":7}]},{"BracketExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":7}]},{"DeclBracket":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":7}]},[{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":12,"srcSpanEndLine":12,"srcSpanStartColumn":5},"srcInfoPoints":[]},7,"7"]}]}]}]]}]}]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":9}]},{"BracketExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":9}]},{"TypeBracket":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":9}]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":14,"srcSpanEndLine":14,"srcSpanStartColumn":5},"srcInfoPoints":[]},"Int"]}]}]}]}]}]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":7}]},{"BracketExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":7}]},{"PatBracket":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"PLit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":16,"srcSpanEndLine":16,"srcSpanStartColumn":5},"srcInfoPoints":[]},7,"7"]}]}]}]}]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":17}]},{"SpliceExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":17}]},{"ParenSplice":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":3},"srcInfoPoints":[]},"deriveStuff"]}]}]},{"VarQuote":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":15},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":15}]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":16},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":18,"srcSpanEndLine":18,"srcSpanStartColumn":16},"srcInfoPoints":[]},"f"]}]}]}]}]}]}]},{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":2,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":2,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":1},"srcInfoPoints":[]},"m"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":3},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":3}]},{"SpliceExp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"IdSplice":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":19,"srcSpanEndLine":19,"srcSpanStartColumn":5},"srcInfoPoints":[]},"x"]}]}]},null]},{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":1}]},{"TypQuote":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":1}]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":21,"srcSpanEndLine":21,"srcSpanStartColumn":3},"srcInfoPoints":[]},"T"]}]}]}]},{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":1},"srcInfoPoints":[]},[{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":2,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":1},"srcInfoPoints":[]},"f"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":3},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":3},"srcInfoPoints":[]},"n"]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":5},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":5}]},{"Lambda":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":7},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":7},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":22}]},[{"PQuasiQuote":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":9},"srcInfoPoints":[]},"haskell","y"]}],{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":25},"srcInfoPoints":[]},"y"]}]}]},{"QVarOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":26},"srcInfoPoints":[]},"+"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":23,"srcSpanEndLine":23,"srcSpanStartColumn":27},"srcInfoPoints":[]},"n"]}]}]}]}]}]},null]}]]}]]},[]]