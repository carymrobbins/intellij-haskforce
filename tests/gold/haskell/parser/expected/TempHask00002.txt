Haskell File
  PPRAGMA
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)('LANGUAGE')
    PsiElement(HaskellTokenType.PRAGMA)('FlexibleInstances')
    PsiElement(HaskellTokenType.#-})('#-}')
  PPRAGMA
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)('LANGUAGE')
    PsiElement(HaskellTokenType.PRAGMA)('QuasiQuotes')
    PsiElement(HaskellTokenType.#-})('#-}')
  PPRAGMA
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)('LANGUAGE')
    PsiElement(HaskellTokenType.PRAGMA)('TemplateHaskell')
    PsiElement(HaskellTokenType.#-})('#-}')
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('TempHask00002')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    PsiElement(HaskellTokenType.Quasi-[)('[')
    QVARID
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('undefined')
    PsiElement(HaskellTokenType.|)('|')
    QQBLOB
      PsiElement(HaskellTokenType.Quasi-stuff)('\nsomething here "hello world"\n')
    PsiElement(HaskellTokenType.|])('|]')
    PsiElement(HaskellTokenType.[|)('[|')
    EXP
      PsiElement(HaskellTokenType.integertoken)('7')
    PsiElement(HaskellTokenType.|])('|]')
    PsiElement(HaskellTokenType.Quasi-[)('[')
    QVARID
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('d')
    PsiElement(HaskellTokenType.|)('|')
    QQBLOB
      PsiElement(HaskellTokenType.Quasi-stuff)(' 7 ')
    PsiElement(HaskellTokenType.|])('|]')
    PsiElement(HaskellTokenType.Quasi-[)('[')
    QVARID
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('t')
    PsiElement(HaskellTokenType.|)('|')
    QQBLOB
      PsiElement(HaskellTokenType.Quasi-stuff)(' Int ')
    PsiElement(HaskellTokenType.|])('|]')
    PsiElement(HaskellTokenType.Quasi-[)('[')
    QVARID
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('p')
    PsiElement(HaskellTokenType.|)('|')
    QQBLOB
      PsiElement(HaskellTokenType.Quasi-stuff)(' 7 ')
    PsiElement(HaskellTokenType.|])('|]')
    PsiElement(HaskellTokenType.$()('$(')
    EXP
      QVAR
        QVARID
          VARID
            PsiElement(HaskellTokenType.varidRegexp)('deriveStuff')
      PsiElement(HaskellTokenType.')(''')
      QVAR
        QVARID
          VARID
            PsiElement(HaskellTokenType.varidRegexp)('f')
    PsiElement(HaskellTokenType.))(')')
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('m')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          PsiElement(HaskellTokenType.idsplice)('$x')
    PsiElement(HaskellTokenType.'')('''')
    QCON
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('T')
    FUNORPATDECL
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('f')
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('n')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          PsiElement(HaskellTokenType.\)('\')
          PsiElement(HaskellTokenType.Quasi-[)('[')
          QVARID
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('haskell')
          PsiElement(HaskellTokenType.|)('|')
          QQBLOB
            PsiElement(HaskellTokenType.Quasi-stuff)('y')
          PsiElement(HaskellTokenType.|])('|]')
          PsiElement(HaskellTokenType.->)('->')
          EXP
            QVAR
              QVARID
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('y')
            QOP
              QVAROP
                QVARSYM
                  VARSYM
                    PsiElement(HaskellTokenType.+)('+')
            QVAR
              QVARID
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('n')